codice/ddf
# DDF-2528: Adding AttributeValidator for Location.COUNTRY_CODE

#### What does this PR do?
This PR adds an attribute validator to verify that the Location.COUNTRY_CODE attribute is a valid ISO 3166-1 alpha-3 country code.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@glenhein @jrnorth @peterhuffer 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@kcwire 
@millerw8
#### How should this be tested?
- Run a full build.
- Verify all tests pass (specifically the ValidationParserSpecTest and CountryCodeValidatorTest)
- Test ingestion of a record with an invalid location.country-code:
  - Run local DDF instance.
  - Configure `Metacard Validation Filter Plugin` to not filter metacards with errors/warnings.
  - Configure `Catalog Federation Strategy` to show validation errors.
  - Ingest a record with an invalid `location.country-code` attribute value.
  - Search for the record and verify there is an error message under `Validation Errors` under the `Details` tab.
- Test ingestion of a record with a valid location.country-code:
  - See above, but verify there are no error messages under `Validation Errors` on the metacard.

#### Any background context you want to provide?
The Location.COUNTRY_CODE attribute is now populated by the GeoCoderPlugin. This validation has been added to verify the attribute is populated correctly. See https://github.com/codice/ddf/pull/1171

#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2528
#### Screenshots (if appropriate)
#### Checklist:
